File: benchmark/csharp/Services/FileService.cs
using System.IO;

namespace Benchmark.CSharp.Services
{
    public class FileService
    {
        public string ReadUserFile(string filename)
        {
            // Path Traversal [CWE-22]
            string path = Path.Combine(@"C:\app\uploads\", filename);
            return File.ReadAllText(path);
        }
    }
}

File: benchmark/java/src/main/java/com/benchmark/services/FileService.java
package com.benchmark.services;

import java.io.File;
import java.io.FileInputStream;

public class FileService {
    public void readFile(String filename) {
        try {
            // Path Traversal [CWE-22]
            File file = new File("/var/www/uploads/" + filename);
            FileInputStream fis = new FileInputStream(file);
            fis.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
